pkg/controller/nodeipam/ipam/cidrset/cidr_set_test.go

```diff
@@ -341,6 +343,77 @@ func TestCIDRSet_AllocationOccupied(t *testing.T) {
 	}
 }
 
+func TestDoubleOccupyRelease(t *testing.T) {
+	clusterCIDRStr := "10.42.0.0/16"
+	operations := []struct {
+		cidrStr     string
+		occupy      bool
+		numOccupied int
+	}{
+		{
```

Comment:
this runs sequential, add comments to explain what we expect at each step.
// allocate 1 subnet